src/17D6A0.c

```diff
@@ -24,37 +119,132 @@ INCLUDE_ASM(s32, "17D6A0", func_8024F5AC);
+s32 func_8024F84C(Evt* script, s32 isInitialCall) {
+    s32 arg5;
+    s32 arg4;
+    s32 z;
+    s32 y;
+    s32 x;
+    Bytecode* args;
+
+    args = script->ptrReadPos;
+    x = evt_get_variable(script, *args++);
+    y = evt_get_variable(script, *args++);
+    z = evt_get_variable(script, *args++);
+    arg4 = evt_get_variable(script, *args++);
+    arg5 = evt_get_variable(script, *args++);
+    func_8024EFE0(x, y, z, arg4, arg5, evt_get_variable(script, *args++));
+    return ApiStatus_DONE2;
+}
 
-INCLUDE_ASM(s32, "17D6A0", func_8024F940);
+ApiStatus func_8024F940(Evt* script, s32 isInitialCall) {
+    func_8024F7C8();
+    return ApiStatus_DONE2;
+}
 
-INCLUDE_ASM(s32, "17D6A0", btl_show_battle_message);
+void btl_show_battle_message(s32 messageIndex, s32 duration) {
+    BattlePopup* popup = btl_create_popup();
+    if (popup != NULL) {
+        popup->unk_04 = &func_8024FB3C;
+        popup->drawFunc = &btl_show_message_popup;
+        popup->unk_00 = 0;
+        popup->unk_08 = 0;
+        popup->messageIndex = messageIndex;
+        popup->duration = duration;
+        popup->unk_16 = 0;
+        popup->unk_17 = 1;
+        popup->message = NULL;
+        D_8029F640 = 0;
+        D_802838F8 = popup;
+        D_8029F64A = 0;
+        D_8029F64C = 0;
+        D_8029F64E = 0;
+        D_8029F650 = 0;
+    }
+}
 
-INCLUDE_ASM(s32, "17D6A0", btl_show_variable_battle_message);
+void btl_show_variable_battle_message(s32 messageIndex, s32 duration, s32 varValue) {
+    BattlePopup* popup = btl_create_popup();
+    if (popup != NULL) {
+        popup->unk_04 = &func_8024FB3C;
+        popup->drawFunc = &btl_show_message_popup;
+        popup->unk_00 = 0;
+        popup->unk_08 = 0;
+        popup->messageIndex = messageIndex;
+        popup->duration = duration;
+        popup->unk_16 = 0;
+        popup->unk_17 = 1;
+        popup->message = NULL;
+        D_8029F640 = varValue;
+        D_802838F8 = popup;
+        D_8029F64A = 0;
+        D_8029F64C = 0;
+        D_8029F64E = 0;
+        D_8029F650 = 0;
+    }
+}
 
-INCLUDE_ASM(s32, "17D6A0", btl_is_popup_displayed);
+s32 btl_is_popup_displayed(void) {
+    return D_802838F8 != NULL;
+}
 
-INCLUDE_ASM(s32, "17D6A0", btl_set_popup_duration);
+#ifdef NON_MATCHING // requires data migration
+void btl_set_popup_duration(s32 duration) {
+    BattlePopup* popup = D_802838F8;
+    if (D_8029F64A != NULL && popup != NULL) {
+        popup->duration = duration;
+    }
+}
+#else
+void btl_set_popup_duration(s32 duration);
+INCLUDE_ASM(void, "17D6A0", btl_set_popup_duration);
+#endif
 
 void func_8024FAE8(void) {
     D_8029F64C = 1;
 }
 
-INCLUDE_ASM(s32, "17D6A0", func_8024FAFC);
+void func_8024FAFC(void) {
+    D_8029F64C = 0;
+}
 
-INCLUDE_ASM(s32, "17D6A0", close_action_command_instruction_popup);
+void close_action_command_instruction_popup(void) {
+    BattlePopup* popup = D_802838F8;
+    if (popup != NULL) {
+        if (popup->messageIndex >= 67) {
```

Comment:
can these two ifs be combined into an or? Also could you invert the condition instead of having the return?

Reply:
Everything I tried to clean this up didn't match, including inverting the conditions. It's really weird.

Reply:
Oh nevermind, combining all 3 of these if's matches lol